### PR TITLE
Fix #349

### LINKED ISSUE #349: Validation data aggregates into training data for multi-drug models
I am not fully sure about the consequences of this bug yet (still running the benchmark with a fixed setup). Still I already wanted to share this:

My setup (the DrugANVAEModel is a custom model I am developing, but this does not matter too much here):

```python
from drevalpy.experiment import drug_response_experiment
from drevalpy.visualization.create_report import create_report
from drevalpy.models import MODEL_FACTORY
from drevalpy.datasets import AVAILABLE_DATASETS

import torch

torch.set_float32_matmul_precision('medium')

dataset = "CTRPv2"
run_id = "drugANVI"
test_mode = "LCO"
n_cv_splits = 5
hyperparameter_tuning = False
path_data = "data"
path_out = "results"

baselines = [
    'NaivePredictor',
    'NaiveTissueMeanPredictor',
    'NaiveTissueDrugMeanPredictor',
]

models = [
    'SimpleNeuralNetwork',
    'ElasticNetModel',
    'PCAElasticNet',
    'ScVIElasticNet',
    'RandomForest',
    'PCARandomForest',
    'ScVIRandomForest',
    'DrugANVAEModel'
]

drug_response_experiment(
            models=[MODEL_FACTORY[model] for model in models],
            baselines=[MODEL_FACTORY[baseline] for baseline in baselines],
            response_data=AVAILABLE_DATASETS[dataset](path_data="data"),
            n_cv_splits=n_cv_splits,
            test_mode=test_mode,
            run_id=run_id,
            path_data=path_data,
            path_out=path_out,
            hyperparameter_tuning=hyperparameter_tuning,
            wandb_project=run_id
)

create_report(run_id=run_id, dataset=dataset)
```

What I did:

1. Run the experiment 'from scratch'
2. Delete the DrugANVAEModel from the results directory
3. Re-run the script (all models except DrugANVAEModel are cached)

What I observe:

During the initial training (from scratch), the DrugANVAEModel trains for 500+ steps per epoch, in the second training it trains only for 300+ steps per epoch. Also the convergence in the initial run is way smoother and reaches a loss of 0.7, whereas in the second run the best I get is 1.7 with very shaky convergence. (I made this observations using weights and biases).

Investigating potential reasons:

It seems like [`get_datasets_from_cv_split()`](https://github.com/daisybio/drevalpy/blob/587cefd8e3c60151e85ee311d2f36d7e7b5a4a47/drevalpy/experiment.py#L1259) returns direct references instead of copies for multi-drug models. This is problematic as [here](https://github.com/daisybio/drevalpy/blob/587cefd8e3c60151e85ee311d2f36d7e7b5a4a47/drevalpy/experiment.py#L224), the validation data is added to the training data. Now when training many models, the validation data is actually added to the training data multiple times, resulting in my observed 500+ steps per epoch compared to 300+ when only one model is actually trained (and the rest is cached).

### PR DESCRIPTION
**Pull request by nictru**
Fixes #349

### CODE REVIEW

**Comment by nictru** [comment]
I am currently checking how this could be tested

**Review by PascalIversen — APPROVED** [review]
the tests are great, thanks! looks good to me!
